File: biloop_integration/parse_invoices.js
const fs = require('fs');

let content = fs.readFileSync('api_v1.json', 'utf8').trim();
const lastIndex = content.lastIndexOf('}');
content = content.substring(0, lastIndex + 1);

let data;
try {
    data = JSON.parse(content);
} catch (e) {
    const fallback = content.substring(0, e.position || content.length);
    console.log("Error parsing JSON");
}

if(data) {
  const schema = data.paths['/erp/incomes/invoices/postInvoices'].post.requestBody.content['application/json'].schema.items.properties;
  
  console.log("Check for tax included or vat flags:");
  for (const [k, v] of Object.entries(schema)) {
      if (typeof v === 'object' && v !== null) {
          if(k.includes('tax') || k.includes('vat') || k.includes('inc'))
          console.log(`${k}: ${v.description ? v.description.substring(0, 100) : ''}`);
      }
  }
}

File: biloop_integration/parse_invoices.py
import json

with open("api_v1.json", "r") as f:
    text = f.read().strip()
    
try:
    data = json.loads(text)
except json.JSONDecodeError as e:
    text = text[:e.pos]
    text = text[:text.rfind('}')+1] + "}"
    data = json.loads(text)

schema = data["paths"]["/erp/incomes/invoices/postInvoices"]["post"]["requestBody"]["content"]["application/json"]["schema"]

props = schema.get("items", {}).get("properties", {})
print("--- TOP LEVEL ---")
for k, v in props.items():
    print(f"{k}: {v.get('type')} - {v.get('description', '')[:100]}")

line_props = props.get("ERP_line", {}).get("items", {}).get("properties", {})
print("\n--- LINE ITEMS ---")
for k, v in line_props.items():
    print(f"{k}: {v.get('type')} - {v.get('description', '')[:100]}")
